### PR TITLE
Add CLI for cloudkeeper-aws

### PR DESCRIPTION
**Pull request by dudoslav**
CLI includes method sync and version. Also this commit adds THOR into
dependencies, gem bin folder and also add config file.

### CODE REVIEW

File: lib/cloudkeeper/aws/cli.rb [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+module Cloudkeeper
+  module Aws
+    # Class defining CLI of cloudkeeper-aws
+    class CLI < Thor
+      SIGINT = 2
+      SIGTERM = 15
+      SIGNALS = [SIGTERM, SIGINT].freeze
+
+      method_option :'polling-timeout',
```
Comment (amarthadan) [inline lib/cloudkeeper/aws/cli.rb:11]: Missing option description.

File: lib/cloudkeeper/aws/cli.rb [hunk 2]
```diff
@@ -0,0 +1,93 @@
+require 'thor'
+
+module Cloudkeeper
+  module Aws
+    # Class defining CLI of cloudkeeper-aws
+    class CLI < Thor
```
Comment (amarthadan) [inline lib/cloudkeeper/aws/cli.rb:6]: Missing options for logging and debugging. Is that on purpous? Will that be included in the logging PR?

File: lib/cloudkeeper/aws/cli.rb [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      desc 'version', 'Prints Cloudkeeper-AWS version'
+      def version
+        $stdout.puts Cloudkeeper::Aws::VERSION
+      end
+
+      default_task :sync
+
+      private
+
+      def initialize_cloud
```
Comment (amarthadan) [inline lib/cloudkeeper/aws/cli.rb]: What is the point of this method and keeping `Cloudkeeper::Aws::Cloud` as an instance variable? Wouldn't be easier to call `Cloudkeeper::Aws::CoreConnector.new(Cloudkeeper::Aws::Cloud.new)` directly?

File: lib/cloudkeeper/aws/cli.rb (hunk 1 above)
Comment (amarthadan) [inline lib/cloudkeeper/aws/cli.rb:11]: `cloudkeeper-one` is using gem `chronic_duration` to parse human readable time periods. Maybe you should considre doing the same.

File: lib/cloudkeeper/aws/cli.rb [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+                    default: Cloudkeeper::Aws::Settings['polling_interval'],
+                    type: :numeric
+      method_option :'bucket-name',
+                    default: Cloudkeeper::Aws::Settings['bucket_name'],
+                    type: :string
+      method_option :'listen-address',
+                    default: Cloudkeeper::Aws::Settings['listen-address'],
+                    type: :string,
+                    desc: 'IP address gRPC server will listen on'
+      method_option :authentication,
```
Comment (amarthadan) [inline lib/cloudkeeper/aws/cli.rb:27]: You should add a check which will see whether when `authentication` option is set to `true` options `certificate`, `key` and `core-certificate` are set as well. This way `TypeError: no implicit conversion of nil into String` would be thrown and that is quite misleading.

Take a look at [https://github.com/the-cloudkeeper-project/cloudkeeper-one/blob/master/lib/cloudkeeper/one/cli.rb#L133](https://github.com/the-cloudkeeper-project/cloudkeeper-one/blob/master/lib/cloudkeeper/one/cli.rb#L133).

File: config/cloudkeeper-aws.yml [hunk 5]
```diff
@@ -1,5 +1,16 @@
...
+certificate: /etc/grid-security/hostcert.pem # Backend's host certificate
+key: /etc/grid-security/hostkey.pem # Backend's host key
+identifier: cloudkeeper-aws # Instance identifier
+core:
+  certificate: /etc/grid-security/corecert.pem # Core's certificate
+logging:
+  level: ERROR # Logging level
+  file: /var/log/cloudkeeper/cloudkeeper-aws.log # File to write log to. To turn off file logging leave this field empty.
+debug: false # Debug mode
 aws:
```
Comment (amarthadan) [inline config/cloudkeeper-aws.yml:15]: What exactly can be set in this section? I thought that we would leave that for the AWS configuration file.

**Review by amarthadan — CHANGES_REQUESTED** [review]

File: lib/cloudkeeper/aws/cli.rb (hunk 1 above)
Comment (dudoslav) [reply lib/cloudkeeper/aws/cli.rb:11]: I considered it

**Review by amarthadan — APPROVED** [review]
:eagle:

**Pull request merged**